### DISCUSSION TITLE
Home Assistant Sensor Display Possible Issue

### DISCUSSION THREAD

**Discussion by lloydbayley** [Nachtzuster/BirdNET-Pi#20] [General]
After migrating to this repository, there is a very small issue that arose, at least with my installation, this is why I created a discussion and not an issue, because it is beyond the initial intent/purpose.

I have a Home Assistant installation that shows me the last bird detected with the details of the detection in a particular card-arrangement.

This stopped working after I migrated and today, I found the cause. It happened because the new and improved way of parsing the MQTT body string and it's not an 'issue' because the new fixes have made the change needed, permanent. Which is an excellent thing and this can be rectified very easily.

The original configuration was from @phillprice in a post that was on the original repo but he has memorialised in his blog here: [https://www.phillprice.com/post/show-birdpi-detections-in-home-assistant/](https://www.phillprice.com/post/show-birdpi-detections-in-home-assistant/)

The alteration I needed to make due to the fact that my JSON MQTT string had lost the double-quotes required and thus made it an erroneous JSON string when it was attempted to read in to HA.

What was in my birdnet.conf file after I migrated:
```
{Common_Name: $comname, Scientific_Name: $sciname, Confidence_Score: $confidence, link: $listenurl, Date: $date, Time: $time, Week: $week, Latitude: $latitude, Longitude: $longitude, Minimum_Confidence: $cutoff, Sigmoid_Sensitivity: $sens, Overlap: $overlap, Image: $flickrimage}
```

What I SHOULD have had:
```
{"Common_Name": "$comname", "Scientific_Name": "$sciname", "Confidence_Score": "$confidence", "link": "$listenurl", "Date": "$date", "Time": "$time", "Week": "$week", "Latitude": "$latitude", "Longitude": "$longitude", "Minimum_Confidence": "$cutoff", "Sigmoid_Sensitivity": "$sens", "Overlap": "$overlap", "Image": "$flickrimage"}
```

As you can see, the double-quotes around the fields somehow were lost in translation and needed to be re-established.

It's possible it was also something to do with my initial setup many, many moons ago when I first installed BirdNetPi, however, it may prove useful to others, should you experience the same symptoms! :)

All the very best,

Lloyd.

**Comment by Nachtzuster** [comment]
For completeness: when you set this though the ui, as one can and should these days, the string will be 'escaped' and looking like this in `birdnet.conf`:
```
"{&#34;Common_Name&#34;: &#34;$comname&#34;, &#34;Scientific_Name&#34;: &#34;$sciname&#34;, &#34;Confidence_Score&#34;: &#34;$confidence&#34;, &#34;link&#34;: &#34;$listenurl&#34;, &#34;Date&#34;: &#34;$date&#34;, &#34;Time&#34;: &#34;$time&#34;, &#34;Week&#34;: &#34;$week&#34;, &#34;Latitude&#34;: &#34;$latitude&#34;, &#34;Longitude&#34;: &#34;$longitude&#34;, &#34;Minimum_Confidence&#34;: &#34;$cutoff&#34;, &#34;Sigmoid_Sensitivity&#34;: &#34;$sens&#34;, &#34;Overlap&#34;: &#34;$overlap&#34;, &#34;Image&#34;: &#34;$flickrimage&#34;, &#34;Reason&#34;:  &#34;$reason&#34;}"
```

**Comment by lloydbayley** [comment]
Agreed. All should be done through the UI. Was just pointing out what transpired in my case as it's been a problem in the old repo for some time now.
